modules/ztp-image-based-upgrade.adoc

```diff
@@ -0,0 +1,106 @@
+
+[IMPORTANT]
+====
+The image-based upgrade uses custom images that are specific to the hardware platform that the clusters are running on.
+Each different hardware platform requires a separate seed image.
+====
+// TODO agree on how much we should specify this. Is this enough or detail about CPU topology, deployment method, etc.
+
+The {lcao} uses two custom resources (CRs) on the participating clusters:
+
+* On the seed cluster, the `SeedGenerator` CR allows the seed image generation. This CR specifies the repository to push the seed image to.
+* On the target cluster, the `ImageBasedUpgrade` CR specifies the seed container image for the upgrade of the target cluster and the backup configurations for your workloads.
+
+.Example SeedGenerator CR
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: SeedGenerator
+metadata:
+  name: seedimage
+spec:
+  seedImage: <seed_container_image>
+----
+
+.Example ImageBasedUpgrade CR
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: ImageBasedUpgrade
+metadata:
+  generation: 1
+  name: upgrade
+spec:
+  stage: Idle
+status:
+  conditions:
+  - lastTransitionTime: "2024-01-19T05:42:58Z"
+    message: Idle
+    observedGeneration: 1
+    reason: Idle
+    status: "True"
+    type: Idle
+  observedGeneration: 1
+----
+
+After generating the seed image on the seed cluster, you can move through the stages on the target cluster by specifying the image to pull and then setting the `spec.stage` field to the following values in the `ImageBasedUpgrade` CR:
+
+* `Idle`
+* `Prep`
+* `Upgrade`
+* `Rollback` (Optional)
+* `Idle`
+
+[discrete]
+== Idle stage
+
+This is the default stage when the {lcao} is installed. There is no ongoing upgrade and the cluster is ready to move to the `Prep` stage.
+
+After a successful upgrade or a rollback, you commit to the change by patching the `stage` field to `Idle` in the `ImageBasedUpgrade` CR.
+Changing to this stage ensures that the {lcao} cleans up resources, so the cluster is ready for upgrades again.
+
+[discrete]
+== Prep stage
+
+You can complete this stage before a scheduled maintenance window.
+
+In the `Prep` stage, the {lcao} ensures that the target cluster is ready to proceed to the `Upgrade` stage by checking if it meets certain conditions.
+The Operator uses the `ostree` versioning system and the OADP Operator to create a backup, which allow complete cluster reconfiguration after upgrade or rollback.
+
+At the `Prep` stage, the {lcao} pulls the seed image and additional container images specified in the seed image to the target cluster.
+
+You can stop the upgrade process at this point by patching the `stage` field to `Idle` in the `ImageBasedUpgrade` CR.
+If you stop, the Operator performs cleanup operations.
+
+[discrete]
+== Upgrade stage
+
+Before the {lcao} starts the upgrade process, you can create an optional backup of your cluster resources with the OADP Operator.
```

Comment:
In the Prep stage (:81) also says about preparing backup. Where is the backup really done?

Reply:
ok, reading further, it seems that on "prep" you specify what do you want to backup. Then on "upgrade" the backup is done. Would this be more clear in the definition of both stages?